privacyidea/api/lib/postpolicy.py

```diff
@@ -446,10 +454,11 @@ def save_pin_change(request, response, serial=None):
                 # anew.
                 pinpol = policy_object.get_action_values(
                     ACTION.CHANGE_PIN_EVERY, scope=SCOPE.ENROLL,
-                    realm=realm, client=g.client_ip, unique=True)
+                    realm=realm, client=g.client_ip, unique=True,
+                    audit_data=g.audit_object.audit_data)
                 if pinpol:
                     token = get_tokens(serial=serial)[0]
-                    token.set_next_pin_change(diff=pinpol[0])
+                    token.set_next_pin_change(diff=pinpol.keys()[0])
```

Comment:
Please switch to `diff=list(pinpol)[0]` in order to be compliant with Python 3.
Please keep in mind, that the order of items in dictionaries is not guaranteed, especially in Python 3.

Reply:
I will change all occurrances.
The nice thing is, that in such cases the list only has one element.

Reply:
done. please check the upcoming commit 39f2f33c4a54e7fc9353453dfddcce6bb1b52d98

Reply:
looks good.